ksql-metastore/src/main/java/io/confluent/ksql/metastore/MetaStore.java

```diff
@@ -43,5 +45,19 @@
 
   public void putAll(MetaStore otherMetaStore);
 
+  public void addSourceNames(Set<String> sourceNames, String queryId);
```

Comment:
see comment above - probably can collapse these two methods into one, i.e, 
`addPersistentQuery(queryId, sources, sinks)`
Also - we should be using `QueryId` rather than `String`